fixes #214 #220

t0rrant opened the pull request:
## Description

Mainly fixes setting the root password in the server_install recipe. The way this is achieved is by first killing off any mysql service that may be running (and which are properly registered in /var/run/mysql/mysqld.pid), launching up a mysqld service manually, with an initial script which sets the root password. Then it tries to authenticate with that password, and if it is okay, it kills that process in order to allow the proper mariadb service to start.

Properly checks for the `node['mariadb']['server_root_password']` attribute being set, as well as the `password` property being passed. 

Fixes errors about the way the `only_if` guard is evaluated, it is now evaluated as a code block.

Also removed a reference in the integration tests about the `reload` function for mariadb service, which does not support this function.

### Issues Resolved

#214 #220 

### Check List
- [x] All tests pass. See https://github.com/sous-chefs/apache2/blob/master/TESTING.md
  tested with kitchen in Ubuntu 16.04 and 18.04, debian doesn't pass tests due to another issue with importing apt-keys (see chef/chef#7913)
- [x] New functionality includes testing.
  the testing is the recipe not failing by itself

t0rrant committed:
fixes #214 #178
fixes `only_if` and mysql query issues with setting password

t0rrant committed:
fixes #214 #178
before starting the service, checks for an initial config file, creates it if it doesn't exist and runs mysqld with init-file argument

t0rrant committed:
fixes #214 #178
minor fix

t0rrant committed:
fixes #214 #178
minor fix, file has :create, not :run...

t0rrant committed:
fixes #214
path got lost somewhere in time

t0rrant committed:
fixes #214
Updated mysqld command to run as root for setting the initial password as per https://dev.mysql.com/doc/refman/5.7/en/changing-mysql-user.html

t0rrant committed:
fixes #214
updates root password, and verifies it was updated

t0rrant committed:
fixes #214
added failsafe to make sure the mysql service is not running

t0rrant committed:
fixes #214
added failsafe to make sure the mysql service is not running

t0rrant committed:
fixes #214
added failsafe to make sure the mysql service is not running

t0rrant committed:
fixes #214
removed reload support as mariadb service does not support it and can be misleading

t0rrant committed:
fixes #214
added presence test for pid_file

t0rrant committed:
fixes #214
fixing issues with kill

t0rrant committed:
fixes #214
fixing issues using kill

t0rrant committed:
fixes #214
apparently the mysqld service is stopped afterwards

t0rrant committed:
fixes #214
improved resource notifications

t0rrant committed:
fixes #214
improved resource notifications

t0rrant committed:
fixes #214
removed `only_if` clause, as `generate-mariadb-root-password` resource already creates the file for us and is required by this resource

t0rrant committed:
fixes #214
removed `only_if` clause, as `generate-mariadb-root-password` resource already creates the file for us and is required by this resource

t0rrant committed:
fixes #214
working out chef resource behaviour...

t0rrant committed:
fixes #214
fixes linting

kitchen-porter commented:
<!--
  1 Error: Please include a <a href="http...
  0 Warnings
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a <a href="https://github.com/sous-chefs/line-cookbook/blob/master/CHANGELOG.md">CHANGELOG</a> entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

t0rrant committed:
fixes #214
bad boolean algebra... it would not apply password if there was any mysqld process to kill

t0rrant commented:
I just realized a big mistake on the boolean algebra, closing another one...
Is it possible to update a PR instead of submitting another one?
